### ISSUE TITLE
Replace MAI's SOLR calls with MAI database queries

### ISSUE THREAD

**Issue by davidskalinder** [davidskalinder/mpeds-coder#71]
I _think_ that there are only two pages that do any such calls: the main article display (which calls SOLR for the article text and the metadata fields along the top); and the admin page (which gets the publication and maybe the database names).  I need to review these pages for details, and I'll try grepping around a bit to make sure I'm not missing anything (and @alexhanna, please feel free to remind me of any others I'm overlooking!).

But whether it's just for those two calls or for more in addition, once #70 gets the necessary info into the MAI DB, those calls should be converted to regular ol' DB queries, with no change apparent to users.

**Comment by davidskalinder** [comment]
```
$ git checkout upstream_development
$ grep -rli --exclude-dir=".git" --exclude="*.pyc" "solr"
generate-coder-table.py
mpeds_coder.py
README.md
config.py-example
templates/event-creator-block.html
templates/admin.html
static/admin.js
config.py
```

**Comment by davidskalinder** [comment]
#### Files that need fixing
```
mpeds_coder.py
```
This is the big one, of course.  And, it turns out, probably the only one.

Of course there are also files in other branches to handle analysis and ETL of Solr articles, but they're just tools to help get rid of the thing so shouldn't need fixing (and indeed can be deleted once everything's in MAI).

#### Files that should probably be tidied eventually
```
README.md
config.py-example
config.py
```
None of these do anything on their own of course, as well they shouldn't.  I guess it'd be sensible to remove the Solr settings in `config.py-example` (which unlike the real config is under version control).  And of course we could clean up the readme, although there's probably a lot of other wrong stuff in there that should be cleaned up someday anyway, to appease the masses of users wanting to install MAI on their own.

#### Files that could be tidied but maybe shouldn't be
```
templates/admin.html
static/admin.js
```
Looks like these both just do the Solr-search-syntax generator, which I believe doesn't actually talk to Solr but requires the user to paste the generated syntax into another Solr interface.  So no change necessary here really.  (We could take it out, though on the other hand, hey, free Solr-search-syntax generator.  And I think, e.g., @olderwoman uses this from time to time and still might to check articles not in MAI, even after Solricide.)

#### Files that mention Solr but don't require action
```
generate-coder-table.py
```
Looks like this only names a field of its output `solr_id`.  So probably nothing to do here.

```
templates/event-creator-block.html
```
Just a comment mentioning a dream of a future feature to search Solr for past and related events.  Probably nothing to do here.

**Comment by davidskalinder** [comment]
Okay so I should note that coming up is the point where we remove this line:

https://github.com/davidskalinder/mpeds-coder/blob/b5f10df8bb58eb038be5fa86ce679a31e6f9092e/mpeds_coder.py#L130

... and therefore lose section, byline, and dateline from the MAI article display since unlike publication, date, and the Solr ID, they won't be in MAI's DB.

This doesn't affect our project since as noted in https://github.com/davidskalinder/mpeds-coder/issues/59#issuecomment-610646341, none of our the articles in our MAI DBs have a section (that we care about), byline, or dateline in Solr; #103 is open to eventually identify where the only one of these we even slightly care about (author/byline) lives and get it into MAI.  But that's not a very high priority for us.

So @alexhanna, if you lot care about those fields, it's nearly time to speak or forever hold your peace (or fix your peace upstream)!

**Comment by davidskalinder** [comment]
Within `mpeds_coder.py`, Solr appears in a handful of functions, some more expected than others:

`loadSolr()`: A helper function that calls up Solr with a Solr ID and returns the article title, specified metadata, and a list of paragraphs.  Looks like it's only called once, in `prepText()`.

`prepText()`: This is the big expected one.  Calls `loadSolr()` and then does a bunch of stuff with the article title, metadata, and paragraphs it gets back.

`convertIDToPublication()`: Extracts the pub name out of a Solr ID, by position it looks like.  Called only once, in `_pubCount()`, the only call to which is commented out and which has a note in its header saying that it should be reimplemented in pandas.  So, looks like this both `convertIDToPublication()` and `_pubCount()` could be removed altogether!  Though not sure whether the Solricide branch is the best place for it or not...

`admin()`: It looks like this function queries Solr directly to do something having to do with article assignments that I don't understand after a quick skim.  @alexhanna, it looks like you might be using this actively, since there's a University Wire url hard-coded in there -- maybe you can advise on what this does?  Anyway, I'll be careful with this and not change it until I fully understand it.

`generateCoderAudit()`: Looks like this just gets the `db_id` column from the MAI database and names it `solr_id` for the `coder-table` export.  Probably fine to leave it until and unless we rename all the columns named `solr_id`.

`searchSolr()`: Runs the Solr-search-syntax generator mentioned above that doesn't actually call Solr.  Probably fine to leave this alone.

----
So it looks like this adds up to three areas of action:

1. Have `prepText()` query MAI's DB instead of Solr, and do everything else the same way.  Delete `loadSolr()`.
2. Understand how `admin()` uses Solr info and get the same info from an MAI DB query.
3. Optionally, get rid of `convertIDToPublication()` and `_pubCount()`.

I'll open a separate issue for that last one since it doesn't affect the rest of the stuff and should probably get PR'd separately anyway.

@alexhanna, let me know if it's immediately clear to you what I should do with that Solr call in the `admin()` function since that will speed up my code-reading.

**Comment by davidskalinder** [comment]
>     1. Have `prepText()` query MAI's DB instead of Solr, and do everything else the same way.  Delete `loadSolr()`.

Well that turned out to be pretty easy.  I chickened out and left the Solr functionality in in case we need it, but added a new config option to use internal storage and override the other storage options (just as the Solr option overrode the older text-file options).  This has the nice effect of being able to toggle the config options to test what has(n't) changed.

Commit 69fb00d5f9 should do the trick and is working in the development deployment.  I'll quote from the commit message:

>     Add ability for prepText() to get all article info from internal DB
>    
>    Requires a new entry in config.py setting STORE_ARTICLES_INTERNALLY
>    to True.
>    
>    Changes display of the old INTERNAL_ID field to display of MAI's db_id
>    field, which has more information and is used as a unique ID elsewhere
>    in MAI and exports.

**Comment by davidskalinder** [comment]
>     2. Understand how `admin()` uses Solr info and get the same info from an MAI DB query.

And I think ff7ed66934 should take care of this in a similar way: if `STORE_ARTICLES_INTERNALLY` is set to `True` in the config, then the admin article assignment page grabs the list of publication from `articles_metadata`; if not, then it'll fall back on getting them from Solr if it can.

However: fundamentally, I still don't understand how this list of publications is used to assign articles, and I've never used that part of the assignment functionality, so it's hard for me to test that this change isn't breaking anything.  (E.g., maybe the chosen pub needs to match something that it won't with this setup?)

I think, then that this is as far as I can go with this one.  So I think the way to go here should be to PR it with the rest of the PRs from #59 (after the next article load is finished) and then for @alexhanna to test it in a non-live deployment to make sure it works as expected.

**Comment by davidskalinder** [comment]
Upstream in https://github.com/MPEDS/mpeds-coder/pull/48.  Deleting branch and closing.
